### PR TITLE
Add types to graph viz code

### PR DESCRIPTION
**Pull request by OskarDamkjaer**

### CODE REVIEW

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:51]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:86]: 👍

File: src/browser/modules/D3Visualization/lib/visualization/components/collision.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/collision.ts:30]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/collision.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/collision.ts:29]: node type is Node & d3.layout.force.Node?

File: src/browser/modules/D3Visualization/lib/visualization/components/layout.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/layout.ts:75]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/renderer.ts
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/renderer.ts:24]: TODO

**Review by eijawerner — COMMENTED** [review]
Nice to see so many types fixed! ⭐ There are a lot of TODO left, I started to mark them out, but finished after a while since they are so many, just do a global search on it :).

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:90]: If d3 changes how it does mutation, might want to compare ids instead here in the future

**Review by eijawerner — APPROVED** [review]
Nice work! Really great to get so many types in, and got rid of a LOT of warnings! 🥇 LGTM when the tests are green.

**Pull request merged**
